src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -501,11 +501,14 @@ bool StorageReplicatedMergeTree::checkFixedGranularityInZookeeper()
 
 
 void StorageReplicatedMergeTree::waitMutationToFinishOnReplicas(
-    const Strings & replicas, const String & mutation_id) const
+    const Strings & replicas, const String & mutation_id)
 {
     if (replicas.empty())
         return;
 
+    /// We need to make sure that local queue has been synced from zookeeper and contains the entry for the
+    /// current mutation. We use local mutation status for checkig for errors.
+    queue.updateMutations(getZooKeeper());
```

Comment:
Maybe we should do it in `while (!partial_shutdown_called)` loop (most likely no)?